### DISCUSSION TITLE
Encrypt / decrypt attributes

### DISCUSSION THREAD

**Discussion by danielbehrendt** [spatie/laravel-data#515] [Q&A]
Is it possible to have a property encrypted when saved to database and decrypted when  fetched?

I already tried using a `Cast` and encryption works so far but I didn't find a way to decrypt the value after data has been set from database.

Is there any way to get this working?

**Comment by rubenvanassche** [comment]
You could use a transformer to encrypt it again, and then a cast to decrypt it. But the transformer will also be executed when running to array on the object and within responses.

Other solution could be to create your own DataEloquentCast and set the property correct over there
